spot-contracts/contracts/BondIssuer.sol

```diff
@@ -140,17 +180,29 @@ contract BondIssuer is IBondIssuer, OwnableUpgradeable {
     /// @dev Lazily issues a new bond when the time is right.
     function getLatestBond() external override returns (IBondController) {
         issue();
-        // NOTE: The latest bond will be at the end of the list.
-        return IBondController(_issuedBonds.at(_issuedBonds.length() - 1));
+        return _lastBond;
     }
 
     /// @inheritdoc IBondIssuer
     function issuedCount() external view override returns (uint256) {
-        return _issuedBonds.length();
+        return _activeBonds.length() + _matureBonds.length();
     }
 
     /// @inheritdoc IBondIssuer
+    /// @dev Internally, we first iterate through the active list and then move on to the mature list.
+    ///      This list is NOT guaranteed to be ordered.
```

Comment:
I guess more specifically:
- `[0, activeBonds.len)` will be ordered by increasing issued date,
-  `[activeBonds.len, numBonds)` will be ordered by increasing maturity date

Reply:
cool. Added this to the comment

Reply:
"[0, activeBonds.len) will be ordered by increasing issued date," -> I dont think this is true.

Since when a bond matures and is removed from the activeBonds list, the last element gets swapped in place messing up the ordering.